Kernel/TTY/TTY.cpp

```diff
@@ -77,7 +126,80 @@ void TTY::emit(u8 ch)
             return;
         }
     }
+
+    if (in_canonical_mode()) {
+        if (is_eof(ch)) {
+            m_available_lines++;
+            //We use '\0' to delimit the end
+            //of a line.
+            m_input_buffer.enqueue('\0');
+            return;
```

Comment:
Please don't put `else` after a `return` statement.
(And the same comment repeated throughout the patch.)